### PR TITLE
Dev cookies banner

### PR DESCRIPTION
**Pull request by msalmeron**
### What does this PR fix/introduce?
Fix GTM data layer not being correctly updated when cookies preferences changes.

### Reviewers
@Adrian-Wrona @andrzej-casper

### CODE REVIEW

**Comment by ghost** [comment]
@msalmeron Please merge latest `dev-new` to fix build. I see that previously `items` were loaded from plugin's data, also there was language considered - how does it work now?

**Comment by msalmeron** [comment]
@andrzej-casper 
There were no changes in how the data is loaded in the modal window. What changed was how we updated the gtm data layer. The data loaded for the UI remains the same.

File: src/theme/DocPage/index.jsx [hunk 1]
```diff
@@ -5,59 +5,12 @@ import { docVersionSearchTag, DocsSidebarProvider, DocsVersionProvider, useDocRo
...
-    const baseUrl = customFields.baseUrl;
-    const path = location.pathname.replace(baseUrl, "");
-    const internalLocale = path.split("/")[0];
-    const externalLocale = useLocaleMap(internalLocale);
-    const data = usePluginData("docusaurus-plugin-cookiesbanner");
 
-    const cookiesData =
-        data?.cookieData.find((x) => x.languageCode === externalLocale) ||
-        data?.cookieData.find((x) => x.languageCode === siteConfig.customFields.defaultExternalLocales);
-    const items = cookiesData?.items;
```
Comment (ghost) [inline src/theme/DocPage/index.jsx:28]: ^ Items are no longer loaded from plugin. I believe that dataLayer will be always empty, but I am unable to test that locally (cookiebar and navdata plugin requires Directus).

@msalmeron If you want, we can merge this PR just to see if it works on staging environment. If it does not, then we will simply revert it.

Comment (msalmeron) [reply src/theme/DocPage/index.jsx:28]: I tested it on my local env. with production builds and it's working. The cookies are first loaded from the modal window, so the data in the cookies will always be present, and the loadtag.js script will update the dataLayer with the data on the cookie "cookies-pref".

Hope this clarify how this is working.

Comment (ghost) [reply src/theme/DocPage/index.jsx:28]: I was worried that cookies from modal will be loaded later than `loadtag.js` script is executed, so I did basic test. Firstly I used static data for cookie modal:
```patch
diff --git a/src/theme/Modals/CookieModal/index.tsx b/src/theme/Modals/CookieModal/index.tsx
index 83b150c66..982d25a37 100644
--- a/src/theme/Modals/CookieModal/index.tsx
+++ b/src/theme/Modals/CookieModal/index.tsx
@@ -76,10 +76,30 @@ function CookieModal() {
     };
 
     const data = usePluginData("docusaurus-plugin-cookiesbanner") as { cookieData: Array<ICookiesData> | undefined };
-
-    const cookiesData =
-        data?.cookieData.find((x) => x.languageCode === externalLocale) ||
-        data?.cookieData.find((x) => x.languageCode === siteConfig.customFields.defaultExternalLocales);
+    let cookiesData: ICookiesData = {
+        languageCode: "a",
+        manage_body: "b",
+        manage_title: "c",
+        notice_body: "d",
+        notice_title: "e",
+        manage_button_text: "f",
+        accept_all_button_text: "g",
+        confirm_button_text: "h",
+        items: [
+            {
+                required: true,
+                parameter: "foo",
+                title: "bar",
+                description: "baz",
+            },
+            {
+                required: false,
+                parameter: "x",
+                title: "y",
+                description: "z",
+            },
+        ],
+    };
 
     useEffect(() => {
         const prefs = Cookies.get("cookie-prefs");
```

and added some basic logs for tracing:

```patch
diff --git a/src/theme/Modals/CookieModal/index.tsx b/src/theme/Modals/CookieModal/index.tsx
index 83b150c66..982d25a37 100644
--- a/src/theme/Modals/CookieModal/index.tsx
+++ b/src/theme/Modals/CookieModal/index.tsx
@@ -116,6 +136,7 @@ function CookieModal() {
         const values = selected.map((x) => {
             return { name: x.name, value: x.value };
         });
+        console.trace("[CookieModal] Setting cookie-prefs");
         Cookies.set("cookie-prefs", JSON.stringify(values));
         window.location.reload();
     };
@@ -124,6 +145,7 @@ function CookieModal() {
         const values = cookiesData?.items.map((x) => {
             return { name: x!.parameter!, value: true };
         });
+        console.trace("[CookieModal] Setting cookie-prefs (all)");
         Cookies.set("cookie-prefs", JSON.stringify(values), { expires: 365 });
         window.location.reload();
     };
diff --git a/static/js/loadtags.js b/static/js/loadtags.js
index 8bd7aeb53..b147e74ea 100644
--- a/static/js/loadtags.js
+++ b/static/js/loadtags.js
@@ -6,6 +6,8 @@ if (cookiePrefs) {
     cookies = JSON.parse(decodeURIComponent(cookiePrefs.split("=")[1]));
 }
 
+console.trace("[loadtags.js] Cookies for dataLayer:", cookies);
+
 if (!window.gtmConfigured && cookies) {
     const layer = "dataLayer";
     window[layer] = window[layer] || [];
@@ -23,4 +25,6 @@ if (!window.gtmConfigured && cookies) {
     });
 
     window.gtmConfigured = true;
+
+    console.trace("[loadtags.js] GTM configured");
 }
```

First page load:
image

After confirming selection page got reloaded:

image

Indeed, it works :+1: I see that `CookieModal` is calling `window.location.reload()` after setting cookies.

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**